apps/etterlatte-saksbehandling-ui/client/src/components/behandling/beregne/Beregne.tsx

```diff
@@ -32,11 +38,43 @@ export const Beregne = (props: { behandling: IBehandlingReducer }) => {
   const [beregning, hentBeregningRequest] = useApiCall(hentBeregning)
   const [vedtak, oppdaterVedtakRequest] = useApiCall(upsertVedtak)
   const [visAttesteringsmodal, setVisAttesteringsmodal] = useState(false)
+  const [, hentSisteIverksatte] = useApiCall(hentSisteIverksatteBehandling)
+  const [, hentBeregningsgrunnlag] = useApiCall(hentBeregningsGrunnlag)
+  const [, kopierGrunnlaget] = useApiCall(kopierBeregningsGrunnlag)
+  const [, postOpprettEllerEndreBeregning] = useApiCall(opprettEllerEndreBeregning)
 
   useEffect(() => {
-    if (!beregningFraState) {
+    const kopierBeregningsgrunnlagHvisOpphoer = async () => {
+      if (behandling.revurderingsaarsak && erOpphoer(behandling.revurderingsaarsak)) {
+        await hentSisteIverksatte(behandling.sak, (res) => {
+          kopierGrunnlagetHvisDetIkkeAltErGjort(res)
+          postOpprettEllerEndreBeregning(behandling.id, () => {
+            dispatch(oppdaterBehandlingsstatus(IBehandlingStatus.BEREGNET))
+          })
+        })
+      }
+    }
+
+    const kopierGrunnlagetHvisDetIkkeAltErGjort = async (forrigeBehandling: IDetaljertBehandling) => {
```

Comment:
Kunne et alternativ vært at backend tok jobben med å sjekke om det var opphør, og evnt da hente foregående grunnlag før den beregner?

Reply:
Absolutt. Spørsmålet er kanskje mest kor ansvaret for denne logikken bør liggje. Men all den tid vi berre kallar _hentBeregning_ akkurat her, kunne det jo vore ei solid forenkling.

Reply:
Okei, det gjorde koden veldig mykje enklare. Gir det eit ordentleg forsøk.